rust/sdk-examples/src/common_steps/latest_processed_version_tracker.rs

```diff
@@ -29,30 +31,59 @@ where
     last_success_batch: Option<TransactionContext<T>>,
     // Tracks all the versions that have been processed out of order.
     seen_versions: AHashMap<u64, TransactionContext<T>>,
+    // Changes behavior given backfill mode.
+    backfill_mode: bool,
+    backfill_start_version: Option<u64>,
+    backfill_end_version: Option<u64>,
 }
 
 impl<T> LatestVersionProcessedTracker<T>
 where
-    Self: Sized + Send + 'static,
+    Self: Sized + Send + Sync + 'static,
```

Comment:
I'm not sure if `Sync` is needed here. It's only to make sure the type is safe to be sent between threads and that isn't necessary here. You might have seen compiler errors with `#[async_trait]` which is unrelated to `Sync` but it's usually means the implementation definition doesn't match the trait.

Reply:
IIUC `Sync` is implicitly derived by the compiler whenever `&self is `Send`. In this case, the compiler is unable to do automatically due to the two asynchronous calls to `save_status` and `save_backfill_status` within `save_processor_status`.

This is the error I get if the explicit `Sync` trait is removed: 
```
future cannot be sent between threads safely
required for the cast from `Pin<Box<{async block@sdk-examples/src/common_steps/latest_processed_version_tracker.rs:232:80: 236:6}>>` to `Pin<Box<dyn futures_util::Future<Output = Result<std::option::Option<Vec<TransactionContext<T>>>, ProcessorError>> + std::marker::Send>>`rustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B3%5D?3#file%3A%2F%2F%2FUsers%2Fseanyang%2Faptos-indexer-processor-sdk%2Frust%2Fsdk-examples%2Fsrc%2Fcommon_steps%2Flatest_processed_version_tracker.rs)
latest_processed_version_tracker.rs(160, 9): captured value is not `Send` because `&` references cannot be sent unless their referent is `Sync`
latest_processed_version_tracker.rs(186, 22): consider further restricting this bound: ` + std::marker::Sync`
```